mule-packager/src/main/java/org/mule/tools/api/util/MavenProjectInformation.java

```diff
@@ -156,13 +164,14 @@ public Pom getEffectivePom() {
   }
 
   private static Optional<ExchangeRepositoryMetadata> getExchangeRepositoryMetadata(DeploymentRepository repository,
-                                                                                    Settings settings) {
+                                                                                    Settings settings,
+                                                                                    List<String> customDomains) {
     ExchangeRepositoryMetadata metadata = null;
-    if (repository != null && ExchangeRepositoryMetadata.isExchangeRepo(repository.getUrl())) {
+    if (repository != null && new ExchangeUriChecker(customDomains).isExchangeRepo(repository.getUrl())) {
```

Comment:
Could we initialize the `ExchangeUriChecker` instance in a different line, It seems too much for an if condition

Reply:
If that makes you happy...